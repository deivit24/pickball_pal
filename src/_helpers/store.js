import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
