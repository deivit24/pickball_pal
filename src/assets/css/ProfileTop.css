#ProfileTop {
  margin-top: -140px;
  height: 500px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background: #3f50b5;
}

#ProfileTop .socials a {
  color: white !important;
  margin: 0px 10px;
}

#GoBack {
  z-index: 100;
  position: fixed;
  top: 100px;
  left: 10px;
}
