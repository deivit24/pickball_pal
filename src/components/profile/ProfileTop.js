import React from 'react';
import '../../assets/css/ProfileTop.css';

const ProfileTop = ({ social }) => {
  return (
    <div id="ProfileTop" className="container-fluid px-0">
      <div className="socials">
        {social && social.facebook && (
          <a href={social.facebook} target="a_blank">
            {' '}
            <i className="fab fa-facebook fa-2x"></i>
          </a>
        )}
        {social && social.instagram && (
          <a href={social.instagram} target="a_blank">
            {' '}
            <i className="fab fa-instagram fa-2x"></i>
          </a>
        )}
        {social && social.linkedin && (
          <a href={social.linkedin} target="a_blank">
            {' '}
            <i className="fab fa-linkedin fa-2x"></i>
          </a>
        )}
      </div>
    </div>
  );
};

export default ProfileTop;
