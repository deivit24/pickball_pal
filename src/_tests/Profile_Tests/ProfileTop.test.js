import React from 'react';
import { render } from '@testing-library/react';
import ProfileTop from '../../components/profile/ProfileTop';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing ProfileTop

describe('Testing the ProfileTop', () => {
  let social = {
    facebook: 'facebook.com',
    instagram: 'instagram.com',
    linkedin: 'linkedin.com',
  };

  it('it renders ProfileTop without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <ProfileTop social={social} />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <ProfileTop social={social} />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
