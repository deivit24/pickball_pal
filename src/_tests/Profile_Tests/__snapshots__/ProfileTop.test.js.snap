// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the ProfileTop matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid px-0"
    id="ProfileTop"
  >
    <div
      class="socials"
    >
      <a
        href="facebook.com"
        target="a_blank"
      >
         
        <i
          class="fab fa-facebook fa-2x"
        />
      </a>
      <a
        href="instagram.com"
        target="a_blank"
      >
         
        <i
          class="fab fa-instagram fa-2x"
        />
      </a>
      <a
        href="linkedin.com"
        target="a_blank"
      >
         
        <i
          class="fab fa-linkedin fa-2x"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
